sympy/printing/ccode.py

```diff
@@ -219,9 +219,36 @@ def _print_ITE(self, expr):
         return self._print(_piecewise)
 
     def _print_MatrixElement(self, expr):
-        return "{0}[{1}]".format(expr.parent, expr.j +
+        from sympy.matrices import MatrixSymbol, MatrixExpr
+
+        if isinstance(expr.parent, MatrixSymbol):
+            return "{0}[{1}]".format(expr.parent, expr.j +
                 expr.i*expr.parent.shape[1])
 
+        terms = (expr.args[0].args)
+        PREC = precedence(expr.args[0])
+        l = []
+        for term in terms:
```

Comment:
I think this code is better off in the `MatrixElement` class. We may want to rewrite:
```python
>>> ((x*A+y*B)[0, 1])._some_expansion_method()
x*A[0, 1] + y*B[0, 1]
```
That is, rewrite the expression by distributing the element access.

I'd suggest to leave this for another PR, so let's (temporarily) ignore the C code printer. If you want, better open a separate PR, this code should be separately tested.